### ISSUE TITLE
package: create a `flux` meta-package

### ISSUE THREAD

**Issue by SteVwonder** [flux-framework/flux-spack#6]
Would be similar to xsdk in structure: https://github.com/spack/spack/blob/3a3f8717185868285ebc1e0226b0d7a320ef404c/var/spack/repos/builtin/packages/xsdk/package.py

Would enable users to run `spack install flux` and get a user-level Flux install versus `spack install flux +system` to get a system-level Flux install (which would include `flux-depend`, `flux-accounting`, etc).

The hard part will be the versioning, since we are a poly-repo without a unified versioning scheme.

**Comment by vsoch** [comment]
Any more discussion around this? Is it a sound idea? Knowing spack, it seems dangerous. :laughing:

**Comment by SteVwonder** [comment]
I don't think we ever packaged the system components like accounting and dependency. It's also not clear to me if anyone would want to install the system version via spack. So maybe not worth pursuing?

**Comment by vsoch** [comment]
ping @trws what say you?

I can confirm that sqlite was missing as a dependency so it's likely that nobody was installing with spack.

**Comment by grondo** [comment]
> I can confirm that sqlite was missing as a dependency so it's likely that nobody was installing with spack.

The strange this is for awhile we did have users installing via spack (not for a system instance of course).  I'm not a spack expert, but I wonder if the spack build could have been finding the system installed sqlite?

**Comment by trws** [comment]
This is the danger of spack's lack of sandboxing. SQLite is almost everywhere, so we only notice the dep is missing when building on a minimal container or somewhere like that where there's no other software. 

As to it being a good idea, I do actually think having a flux metapackage for whatever a "complete" single user flux ought to be would be useful, probably just flux-core and flux-ached for now, but I could see other things in there. System instance would be more packages and more complex, but as @SteVwonder said it probably wouldn't get used all that much, where the user-instance version has several groups using it built through spack.

**Comment by grondo** [comment]
Following up, another good package to add might be [flux-pmix](https://github.com/flux-framework/flux-pmix).

**Comment by trws** [comment]
That's a very good point, I forgot about flux-pmix.
